mount.go

```diff
@@ -91,3 +95,56 @@ func getDeviceForPath(path string) (*device, error) {
 
 	return dev, nil
 }
+
+const (
+	procMountsFile = "/proc/mounts"
+
+	fieldsPerLine = 6
+)
+
+const (
+	procDeviceIndex = iota
+	procPathIndex
+	procTypeIndex
+)
+
+func getDevicePathAndFsType(mountPoint string) (devicePath, fsType string, err error) {
```

Comment:
So there's a kubernetes package for doing a bunch of mount and device operations:
`import "k8s.io/kubernetes/pkg/util/mount"`

And then you can do things like:

```Go
mounter := mount.Mounter{}
mountPoints, err := mounter.List()
```

where `mountPoints` is an array of:

```Go
// This represents a single line in /proc/mounts or /etc/fstab.
type MountPoint struct {
	Device string
	Path   string
	Type   string
	Opts   []string
	Freq   int
	Pass   int
}
```

It also provides some other useful routines like `mounter.PathIsDevice()` or `mount.IsNotMountPoint()` that you could use in your previous patch to reduce code duplication.